aten/src/ATen/native/RNN.cpp

```diff
@@ -1432,10 +1432,16 @@ std::tuple<Tensor, Tensor, Tensor> lstm(
   }
 #ifdef USE_MPS
   if (_input.is_mps() && !bidirectional) {
-    std::tuple<Tensor, Tensor, Tensor, Tensor, Tensor> output = at::_lstm_mps(_input, hx, _params, has_biases,
-            num_layers, dropout_p, train, bidirectional, batch_first);
-    std::tuple<Tensor, Tensor, Tensor> return_values = std::make_tuple(std::get<0>(output), std::get<1>(output), std::get<2>(output));
-    return return_values;
+    // if cells are of different size, that means projections are used
+    bool has_projections = (hx[0].size(2) != hx[1].size(2));
+    if (!has_projections) {
+        std::tuple<Tensor, Tensor, Tensor, Tensor, Tensor> output = at::_lstm_mps(_input, hx, _params, has_biases,
+                num_layers, dropout_p, train, bidirectional, batch_first);
+        std::tuple<Tensor, Tensor, Tensor> return_values = std::make_tuple(std::get<0>(output), std::get<1>(output), std::get<2>(output));
+        return return_values;
+    } else {
+        AT_ERROR("LSTM with projections is not currently supported with MPS.");
```

Comment:
The `ATen/native/RNN.cpp` is a public file, and we should avoid MPS-specific failures in there as much as possible. 
Since you pass `hx` to `at::_lstm_mps()`, can you please check for has_projection inside `_lstm_mps()`, and error out in there instead?

Reply:
I mean you could check for `hx` and error out in here instead:

https://github.com/kulinseth/pytorch/blob/master/aten/src/ATen/native/mps/operations/RnnOps.mm#L27-L30

Reply:
Sure thing @razarmehr! I'll move the check to the MPS specific file instead.

Reply:
Rebased on top of the current master and added the requested change.